workloads/upgrade-perf/rosa_upgrade.sh

```diff
@@ -0,0 +1,139 @@
+  exit 0
+}
+
+rosa_workers_active_waiting(){
+  start_time=$(date +%s)
+  WORKERS=$(oc get node --no-headers -l node-role.kubernetes.io/workload!="",node-role.kubernetes.io/infra!="",node-role.kubernetes.io/worker="" 2>/dev/null | wc -l)
+  # Giving 5 minutes per worker
+  ITERATIONWORKERS=0
+  while [ ${ITERATIONWORKERS} -le $(( ${WORKERS}*5 )) ] ; do
+    if [ $(rosa describe cluster -c ${ROSA_CLUSTER_NAME} | grep ^Scheduled | wc -l) -ne 1 ] ; then
+      echo "INFO: Upgrade finished for ROSA, continuing..."
+      end_time=$(date +%s)
+      export WORKERS_UPGRADE_DURATION=$((${end_time} - ${start_time}))
+      return 0
+    else
+      LASTSTATUS=$(oc logs $(oc get pods -n openshift-managed-upgrade-operator -o Name | grep -v catalog) -n openshift-managed-upgrade-operator | grep "workers are upgraded" | tail -1)
+      echo "INFO: ${ITERATIONWORKERS}/$(( ${WORKERS}*5 )). Last Update: ${LASTSTATUS}."
+      echo "INFO: Waiting 60 seconds for the next check..."
+      ITERATIONWORKERS=$((${ITERATIONWORKERS}+1))
+      sleep 60
+    fi
+  done
+  echo "ERROR: ${ITERATIONWORKERS}/$(( ${WORKERS}*5 )). ROSA workers upgrade not finished after $(( ${WORKERS}*5 )) iterations. Exiting..."
+  end_time=$(date +%s)
+  export WORKERS_UPGRADE_DURATION=$((${end_time} - ${start_time}))
+  oc logs $(oc get pods -n openshift-managed-upgrade-operator -o Name | grep -v catalog) -n openshift-managed-upgrade-operator
+  rosa describe cluster -c ${ROSA_CLUSTER_NAME}
+  return 1
+}
+
+rosa_control_plane_upgrade_active_waiting(){
+  # Giving 90 minutes for controlplane upgrade
+  start_time=$(date +%s)
+  ITERATIONS=0
+  while [ ${ITERATIONS} -le 90 ] ; do
+    VERSION_STATUS=($(oc get clusterversion | sed -e 1d | awk '{print $2" "$3" "$4}'))
+    if [ ${VERSION_STATUS[0]} == $1 ] && [ ${VERSION_STATUS[1]} == "True" ] && [ ${VERSION_STATUS[2]} == "False" ] ; then
+      # Version is upgraded, available=true, progressing=false -> Upgrade finished
+      echo "INFO: OCP upgrade to $1 is finished for OCP, now waiting for ROSA..."
+      end_time=$(date +%s)
+      export CONTROLPLANE_UPGRADE_DURATION=$((${end_time} - ${start_time}))
+      return 0
+    else
+      echo "INFO: ${ITERATIONS}/90. AVAILABLE: ${VERSION_STATUS[1]}, PROGRESSING: ${VERSION_STATUS[2]}. Waiting 60 seconds for the next check..."
+      ITERATIONS=$((${ITERATIONS}+1))
+      sleep 60
+    fi
+  done
+  echo "ERROR: ${ITERATIONS}/90. OCP Version is ${VERSION_STATUS[0]}, not upgraded to $1 after 90 iterations. Exiting..."
+  oc get clusterversion
+  end_time=$(date +%s)
+  export CONTROLPLANE_UPGRADE_DURATION=$((${end_time} - ${start_time}))
+  return 1
+}
+
+rosa_upgrade_index_results(){
+  METADATA=$(grep -v "^#" << EOF
+{
+"uuid" : "${_uuid}",
+"platform": "ROSA",
+"cluster_name": "${ROSA_CLUSTER_NAME}",
+"network_type": "$(oc get network cluster -o json 2>/dev/null | jq -r .status.networkType)",
+"controlplane_upgrade": "$1",
```

Comment:
Can this and the worker_upgrade one be appended with duration? ie workers_upgrade_duration?

Reply:
done